.github/workflows/ci.yml

```diff
@@ -111,11 +81,87 @@ jobs:
           docker run --rm --platform=linux/s390x s390x/ubuntu apt update
           docker run --rm --platform=linux/ppc64le ppc64le/ubuntu apt update
           docker run --rm --platform=linux/arm64 arm64v8/ubuntu apt update
-      - name: Test Syscalls
+      -
+        name: Test Syscalls
         if: matrix.target == 'mainline'
         run: |
           set -x
           docker buildx build --platform=linux/amd64,linux/arm64,linux/386,linux/arm,linux/ppc64le,linux/s390x --target=run --allow security.insecure --build-arg CONFIG_RT_GROUP_SCHED=false ./test
+
+  e2e:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        dockerfile:
+          - pip-youtube-dl
+          - webpack
+        platform:
+          - linux/arm/v7
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Load mainline for testing
+        uses: docker/bake-action@v1
+        with:
+          targets: mainline
+          load: true
+          set: |
+            *.cache-from=type=gha,scope=${{ matrix.target }}
```

Comment:
Not sure I understand what cache this should use. And target is not defined here?

Reply:
My bad yes should be `mainline` as scope.

Reply:
What generates this cache though?

Reply:
Took from `build` job: https://github.com/crazy-max/binfmt/blob/4d57fd5dad7716ed0addf5ac341f7ad1853a4ab3/.github/workflows/ci.yml#L186

Reply:
Yes it does not depend on anything but I thought it would better to let the `build` stage push the cache and let others jobs consume it in the next pipeline trigger. I will create a dedicated scope for each of them. I'm fine with it.

Reply:
If it is a different scope then it is even more likely to miss the cache every time. As most time is spent on qemu binary compilation I don't think reusing cache from master provides much in most cases. What I was getting to was that there should be one job that warms the cache and then this test stage can reuse it.

Reply:
> What I was getting to was that there should be one job that warms the cache and then this test stage can reuse it.

Ah sure we can do that too.

Reply:
Only https://github.com/tonistiigi/binfmt/blob/1aa2eba35192f9e72dd164480ce70cd825469c4b/.github/workflows/ci.yml#L24 will not use any cache atm. But can handle that in a follow-up.